db/system_keyspace.cc

```diff
@@ -2939,12 +2939,6 @@ future<service::topology> system_keyspace::load_topology_state(const std::unorde
         if (row.has("tokens")) {
             auto tokens = decode_tokens(deserialize_set_column(*topology(), row, "tokens"));
 
-            if (tokens.empty()) {
-                on_fatal_internal_error(slogger, format(
```

Comment:
Generally, the whole point of assertions is that they should be impossible to hit :P

The reasoning from commit message is sound, but there could be other reasons why we hit this one:
- bug in `deserialize_set_column`
- bug in `topology()`
- bug in `decode_tokens`

it could be whatever, really.

Reply:
I'm pretty sure this assertion was added to handle the (impossible) case of an empty set. IIRC, I even saw similar assertions in some drivers. It just looks like most people are unaware that an empty set and no value are indistinguishable.

I don't like keeping this assertion for whatever. It looks like it handles the empty set case, but in one of the following patches I explain that there is no such thing as an empty set. It will only cause more confusion.

If we want to have an assertion for whatever, then we should check things like `tokens.size() == num_tokens`. I don't think the current assertion is any useful.

Reply:
OK